.github/workflows/ubuntu.yml

```diff
@@ -37,65 +37,75 @@ jobs:
     steps:
     - uses: actions/checkout@v4
 
-    - name: Cache dependencies
-      uses: actions/cache@v3
-      with:
-        path: |
-          /usr/local/lib
-          /usr/local/include
-          ~/.cache
-        key: ${{ runner.os }}-build-cache-${{ hashFiles('**/CMakeLists.txt') }}
-        restore-keys: |
-          ${{ runner.os }}-build-cache-
+#    - name: Cache dependencies
+#      uses: actions/cache@v3
+#      with:
+#        path: |
+#          /usr/local/lib
+#          /usr/local/include
+#          ~/.cache
+#        key: ${{ runner.os }}-build-cache-${{ hashFiles('**/CMakeLists.txt') }}
+#        restore-keys: |
+#          ${{ runner.os }}-build-cache-
 
     - name: Add Eigen3
-      run: sudo apt install libeigen3-dev
-
-    - name: Add TinyXML2
       run: |
-        sudo apt-get update
-        sudo apt-get install libtinyxml2-dev
-
-    - name: Add urdfom_headers
+        sudo apt update
+        sudo apt install libeigen3-dev
+        
+    - name: Add Rust (Required by Clarabel)
       run: |
-        sudo apt-get update
-        sudo apt-get install -y liburdfdom-headers-dev liburdfdom-tools
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+        echo 'export PATH="$HOME/.cargo/bin:$PATH"' >> ~/.bashrc
+        source ~/.bashrc
+        echo "::add-path::$HOME/.cargo/bin"
 
-    - name: Add console_bridge
+    - name: Add Clarabel
       run: |
-        if [ ! -d "console_bridge" ]
-        then
-          sudo apt-get update
-          sudo apt-get install libconsole-bridge-dev
-        fi
+        sudo apt install cmake -y
+        git clone --recurse-submodules https://github.com/oxfordcontrol/Clarabel.cpp.git
+        cd Clarabel.cpp
+        mkdir build
+        cd build
+        cmake ..
+        cmake --build .
 
-    - name: Clone and build urdfdom
-      run: |
-        if [ ! -d "urdfom" ]
-        then
-          git clone https://github.com/ros/urdfdom.git
-          cd urdfdom
-          mkdir build
-          cd build
-          cmake ..
-          make
-          sudo make install
-        fi
-
-    - name: Add ROS2
+    - name: Add TinyXML2
```

Comment:
What depends on TinyXML2?

Reply:
Some URDF tools, although now that they are commented out this is probably unnecessary as well.